File: sdk/python/examples/cloud.py
from smol import ConnectOptions, Machine, MachineConfig


machine = Machine.create(
    MachineConfig(image="alpine:3.20"),
    ConnectOptions(target="cloud"),  # uses SMOL_CLOUD_TOKEN
)
try:
    # create() waits for ready=True; state "started" only means VM launched.
    result = machine.exec(["echo", "ready for work"]).assert_success()
    print(result.stdout)
finally:
    machine.delete()

File: sdk/node/examples/cloud.ts
import { Machine } from "smolmachines";

async function main() {
  const machine = await Machine.create(
    { image: "alpine:3.20" },
    { target: "cloud" }, // uses SMOL_CLOUD_TOKEN
  );

  try {
    // create() waits for ready=true; state "started" only means VM launched.
    const result = await machine.exec(["echo", "ready for work"]);
    result.assertSuccess();
    console.log(result.stdout);
  } finally {
    await machine.delete();
  }
}

main().catch((error) => {
  console.error(error);
  process.exit(1);
});
